### ISSUE TITLE
Compilation failure on Archlinux

### ISSUE THREAD

**Issue by vmaffione** [the-tcpdump-group/libpcap#656]
Hi,
  Trying to compile the current development code (master branch) on Arch.

System info:
````
$ uname -a
Linux nameXYZ 4.13.5-1-ARCH #1 SMP PREEMPT Fri Oct 6 09:58:47 CEST 2017 x86_64 GNU/Linux
$ gcc --version
gcc (GCC) 7.2.0
$ lscpu
[...]
Model name:          Intel(R) Core(TM) i7-3770K CPU @ 3.50GHz
[...]
````

Steps to reproduce:
````
$ ./configure
$ make
gcc -fvisibility=hidden -fpic -I.  -I/usr/include/dbus-1.0 -I/usr/lib/dbus-1.0/include -DBUILDING_PCAP -DHAVE_CONFIG_H  -D_U_="__attribute__((unused))" -g -O2    -c ./pcap-bpf.c
./pcap-bpf.c: In function 'bpf_open_and_bind':
./pcap-bpf.c:626:16: error: 'BIOCSETIF' undeclared (first use in this function); did you mean 'TIOCSETD'?
  if (ioctl(fd, BIOCSETIF, (caddr_t)&ifr) < 0) {
                ^~~~~~~~~
                TIOCSETD
./pcap-bpf.c:626:16: note: each undeclared identifier is reported only once for each function it appears in
./pcap-bpf.c: In function 'pcap_stats_bpf':
./pcap-bpf.c:903:18: error: storage size of 's' isn't known
  struct bpf_stat s;
                  ^
./pcap-bpf.c:918:19: error: 'BIOCGSTATS' undeclared (first use in this function); did you mean 'SIOCGSTAMP'?
  if (ioctl(p->fd, BIOCGSTATS, (caddr_t)&s) < 0) {
                   ^~~~~~~~~~
                   SIOCGSTAMP
./pcap-bpf.c: In function 'pcap_read_bpf':
./pcap-bpf.c:1105:15: error: dereferencing pointer to incomplete type 'struct bpf_hdr'
   caplen = bhp->bh_caplen;
               ^~
./pcap-bpf.c: In function 'pcap_activate_bpf':
./pcap-bpf.c:1645:21: error: storage size of 'bv' isn't known
  struct bpf_version bv;
                     ^~
./pcap-bpf.c:1677:16: error: 'BIOCVERSION' undeclared (first use in this function); did you mean '_SC_VERSION'?
  if (ioctl(fd, BIOCVERSION, (caddr_t)&bv) < 0) {
                ^~~~~~~~~~~
                _SC_VERSION
./pcap-bpf.c:1683:21: error: 'BPF_MAJOR_VERSION' undeclared (first use in this function); did you mean 'PACKAGE_VERSION'?
  if (bv.bv_major != BPF_MAJOR_VERSION ||
                     ^~~~~~~~~~~~~~~~~
                     PACKAGE_VERSION
./pcap-bpf.c:1684:20: error: 'BPF_MINOR_VERSION' undeclared (first use in this function); did you mean 'BPF_MAJOR_VERSION'?
      bv.bv_minor < BPF_MINOR_VERSION) {
                    ^~~~~~~~~~~~~~~~~
                    BPF_MAJOR_VERSION
./pcap-bpf.c:2013:18: error: 'BIOCSBLEN' undeclared (first use in this function); did you mean 'TIOCSBRK'?
    if (ioctl(fd, BIOCSBLEN,
                  ^~~~~~~~~
                  TIOCSBRK
./pcap-bpf.c:2029:18: error: 'BIOCSETIF' undeclared (first use in this function); did you mean 'TIOCSETD'?
    if (ioctl(fd, BIOCSETIF, (caddr_t)&ifr) < 0)
                  ^~~~~~~~~
                  TIOCSETD
./pcap-bpf.c:2045:19: error: 'BIOCGBLEN' undeclared (first use in this function); did you mean 'BIOCSBLEN'?
    if ((ioctl(fd, BIOCGBLEN, (caddr_t)&v) < 0) ||
                   ^~~~~~~~~
                   BIOCSBLEN
./pcap-bpf.c:2083:16: error: 'BIOCGDLT' undeclared (first use in this function); did you mean 'BIOCGBLEN'?
  if (ioctl(fd, BIOCGDLT, (caddr_t)&v) < 0) {
                ^~~~~~~~
                BIOCGBLEN
./pcap-bpf.c:2380:21: error: 'BIOCSRTIMEOUT' undeclared (first use in this function); did you mean 'BIOCSETIF'?
    if (ioctl(p->fd, BIOCSRTIMEOUT, (caddr_t)&to) < 0) {
                     ^~~~~~~~~~~~~
                     BIOCSETIF
./pcap-bpf.c:2437:20: error: 'BIOCPROMISC' undeclared (first use in this function); did you mean 'IFF_PROMISC'?
   if (ioctl(p->fd, BIOCPROMISC, NULL) < 0) {
                    ^~~~~~~~~~~
                    IFF_PROMISC
./pcap-bpf.c:2496:19: error: 'BIOCSETF' undeclared (first use in this function); did you mean 'BIOCSETIF'?
  if (ioctl(p->fd, BIOCSETF, (caddr_t)&total_prog) < 0) {
                   ^~~~~~~~
                   BIOCSETIF
./pcap-bpf.c: In function 'pcap_setfilter_bpf':
./pcap-bpf.c:3070:19: error: 'BIOCSETF' undeclared (first use in this function); did you mean 'TIOCSETD'?
  if (ioctl(p->fd, BIOCSETF, (caddr_t)fp) == 0) {
                   ^~~~~~~~
                   TIOCSETD
make: *** [Makefile:85: pcap-bpf.o] Error 1
````

**Comment by guyharris** [comment]
It should be fixed in 1d71630de81e0e60fb75e8a24621334294e27ab4; try it with the current tip of the master branch.

**Comment by vmaffione** [comment]
It works now, thanks!
